src/dotnet/commands/dotnet-build3/MSBuildForwardingApp.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class MSBuildForwardingApp : ForwardingApp
+    {
+        private static readonly string s_msbuildExeName = "msbuild.exe";
+
+        public MSBuildForwardingApp(string[] argsToForward) :
+            base(GetMSBuildExePath(), 
+                argsToForward,
+                depsFile: GetDepsFile(),
+                runtimeConfig: GetRuntimeConfig(),
+                environment: GetEnvironment()) { }
+
+        private static Dictionary<string, string> GetEnvironment()
+        {
+            return new Dictionary<string, string>
+            {
+                { "MSBuildExtensionsPath", GetDotnetMSBuildTargetsPath() },
+                { "DotnetHostPath", GetHostPath() }
```

Comment:
See https://github.com/Microsoft/msbuild/pull/681 where we essentially try using the host that msbuild is running under.